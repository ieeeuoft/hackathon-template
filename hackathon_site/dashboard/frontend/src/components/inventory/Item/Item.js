import React from "react";
import Typography from "@material-ui/core/Typography";
import styles from "./Item.module.scss";
import Card from "@material-ui/core/Card";
import CardMedia from "@material-ui/core/CardMedia";

const Item = ({ image, title, total, currentStock }) => {
    const stock = !currentStock
        ? "OUT OF STOCK"
        : currentStock + " OF " + total + " IN STOCK";
    const stockStyle = !currentStock ? styles.outOfStock : styles.inStock;
    const coverStyle = !currentStock ? styles.itemBlack : styles.itemBox;

    return (
        <>
            <Card className={coverStyle} elevation={2} square={true}>
                <CardMedia
                    className={styles.itemPic}
                    component="img"
                    alt={title}
                    image={image}
                />
            </Card>
            <div className={styles.itemTextBox}>
                <Typography variant="body2">{title}</Typography>
                <Typography variant="caption" className={stockStyle}>
                    {stock}
                </Typography>
            </div>
        </>
    );
};

export default Item;
