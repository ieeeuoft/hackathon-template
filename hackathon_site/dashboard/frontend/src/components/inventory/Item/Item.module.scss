@import "./../../../assets/abstracts/mixins";
@import "./../../../assets/abstracts/variables";

.Item {
    display: flex;
    width: 100%;
    flex-direction: column;
    justify-content: space-between;
    align-items: flex-start;
    padding-bottom: 10px;

    &:hover {
        background-color: color(light2);
    }

    &Pic {
        height: 210px;
        &Box {
            object-fit: cover;
            opacity: 0.2;
        }
    }

    &Limit {
        font-size: small;
        margin: -1px 0px 0px -5px;
        &1 {
            color: #98e733;
        }
        &2 {
            color: #dc1c1c;
        }
        &3 {
            color: #ffa456;
        }
        &4 {
            color: #6c36c7;
        }
        &5 {
            color: #c816a1;
        }
    }

    &Name {
        display: flex;
        padding-top: 8px;
    }

    &TextBox {
        padding-left: 10px;
    }

    .outOfStock {
        color: color(red);
    }

    .inStock {
        color: #757575;
    }
}
