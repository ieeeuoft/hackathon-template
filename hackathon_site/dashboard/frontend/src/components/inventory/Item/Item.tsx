import React, { ReactElement } from "react";
import Typography from "@material-ui/core/Typography";
import styles from "./Item.module.scss";
import Card from "@material-ui/core/Card";
import CardMedia from "@material-ui/core/CardMedia";
import { useSelector } from "react-redux";
import { userTypeSelector } from "slices/users/userSlice";

interface ItemProps {
    image: string;
    title: string;
    total: number;
    currentStock: number;
}

const Item = ({ image, title, total, currentStock }: ItemProps): ReactElement => {
    const userType = useSelector(userTypeSelector);
    const stock =
        currentStock < 1
            ? "OUT OF STOCK"
            : userType === "participant"
            ? currentStock + " IN STOCK"
            : currentStock + " OF " + total + " IN STOCK";
    const stockStyle = !currentStock ? styles.outOfStock : styles.inStock;
    const coverStyle = !currentStock ? styles.itemBlack : styles.itemBox;

    return (
        <div className={styles.itemGrid}>
            <Card className={coverStyle} elevation={2} square={true}>
                <CardMedia
                    className={styles.itemPic}
                    component="img"
                    alt={title}
                    image={image}
                />
            </Card>
            <div className={styles.itemTextBox}>
                <Typography variant="body2">{title}</Typography>
                <Typography variant="caption" className={stockStyle}>
                    {stock}
                </Typography>
            </div>
        </div>
    );
};

export default Item;
