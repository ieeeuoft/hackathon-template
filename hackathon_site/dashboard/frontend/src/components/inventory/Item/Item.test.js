import React from "react";
import Item from "./Item";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("InventoryItem", () => {
    const image =
        "https://i.pinimg.com/originals/58/1f/23/581f232c3462392f63d57be5649649de.jpg";
    const title = "Some Hardware";
    const total = 6;
    let limit = null;
    let currentStock = 4;

    test("Has stock with no limit", () => {
        const { asFragment } = render(
            <Item
                image={image}
                title={title}
                total={total}
                limit={limit}
                currentStock={currentStock}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test("Out of Stock", () => {
        let currentStock = null;
        const { asFragment } = render(
            <Item
                image={image}
                title={title}
                total={total}
                limit={limit}
                currentStock={currentStock}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test("limit 1", () => {
        let limit = 1;
        const { asFragment } = render(
            <Item
                image={image}
                title={title}
                total={total}
                limit={limit}
                currentStock={currentStock}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test("limit 2", () => {
        let limit = 2;
        const { asFragment } = render(
            <Item
                image={image}
                title={title}
                total={total}
                limit={limit}
                currentStock={currentStock}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test("limit 3", () => {
        let limit = 3;
        const { asFragment } = render(
            <Item
                image={image}
                title={title}
                total={total}
                limit={limit}
                currentStock={currentStock}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test("limit 4", () => {
        let limit = 4;
        const { asFragment } = render(
            <Item
                image={image}
                title={title}
                total={total}
                limit={limit}
                currentStock={currentStock}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test("limit 5", () => {
        let limit = 5;
        const { asFragment } = render(
            <Item
                image={image}
                title={title}
                total={total}
                limit={limit}
                currentStock={currentStock}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
