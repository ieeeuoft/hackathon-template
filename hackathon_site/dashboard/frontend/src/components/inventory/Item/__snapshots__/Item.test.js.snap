// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryItem Has stock with no limit 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root Item MuiGrid-item"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-outlined"
    >
      <img
        alt="Some Hardware"
        class="MuiCardMedia-root ItemPic MuiCardMedia-media MuiCardMedia-img"
        src="https://i.pinimg.com/originals/58/1f/23/581f232c3462392f63d57be5649649de.jpg"
      />
    </div>
    <div
      class="ItemTextBox"
    >
      <div
        class="ItemName"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Some Hardware
        </p>
      </div>
      <span
        class="MuiTypography-root inStock MuiTypography-caption"
      >
        4 OF 6 IN STOCK
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InventoryItem Out of Stock 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root Item MuiGrid-item"
  >
    <div
      class="MuiPaper-root MuiCard-root ItemPicBox MuiPaper-outlined"
    >
      <img
        alt="Some Hardware"
        class="MuiCardMedia-root ItemPic MuiCardMedia-media MuiCardMedia-img"
        src="https://i.pinimg.com/originals/58/1f/23/581f232c3462392f63d57be5649649de.jpg"
      />
    </div>
    <div
      class="ItemTextBox"
    >
      <div
        class="ItemName"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Some Hardware
        </p>
      </div>
      <span
        class="MuiTypography-root outOfStock MuiTypography-caption"
      >
        OUT OF STOCK
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InventoryItem limit 1 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root Item MuiGrid-item"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-outlined"
    >
      <img
        alt="Some Hardware"
        class="MuiCardMedia-root ItemPic MuiCardMedia-media MuiCardMedia-img"
        src="https://i.pinimg.com/originals/58/1f/23/581f232c3462392f63d57be5649649de.jpg"
      />
    </div>
    <div
      class="ItemTextBox"
    >
      <div
        class="ItemName"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root ItemLimit1"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="8"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Some Hardware
        </p>
      </div>
      <span
        class="MuiTypography-root inStock MuiTypography-caption"
      >
        4 OF 6 IN STOCK
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InventoryItem limit 2 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root Item MuiGrid-item"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-outlined"
    >
      <img
        alt="Some Hardware"
        class="MuiCardMedia-root ItemPic MuiCardMedia-media MuiCardMedia-img"
        src="https://i.pinimg.com/originals/58/1f/23/581f232c3462392f63d57be5649649de.jpg"
      />
    </div>
    <div
      class="ItemTextBox"
    >
      <div
        class="ItemName"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root ItemLimit2"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="8"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Some Hardware
        </p>
      </div>
      <span
        class="MuiTypography-root inStock MuiTypography-caption"
      >
        4 OF 6 IN STOCK
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InventoryItem limit 3 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root Item MuiGrid-item"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-outlined"
    >
      <img
        alt="Some Hardware"
        class="MuiCardMedia-root ItemPic MuiCardMedia-media MuiCardMedia-img"
        src="https://i.pinimg.com/originals/58/1f/23/581f232c3462392f63d57be5649649de.jpg"
      />
    </div>
    <div
      class="ItemTextBox"
    >
      <div
        class="ItemName"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root ItemLimit3"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="8"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Some Hardware
        </p>
      </div>
      <span
        class="MuiTypography-root inStock MuiTypography-caption"
      >
        4 OF 6 IN STOCK
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InventoryItem limit 4 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root Item MuiGrid-item"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-outlined"
    >
      <img
        alt="Some Hardware"
        class="MuiCardMedia-root ItemPic MuiCardMedia-media MuiCardMedia-img"
        src="https://i.pinimg.com/originals/58/1f/23/581f232c3462392f63d57be5649649de.jpg"
      />
    </div>
    <div
      class="ItemTextBox"
    >
      <div
        class="ItemName"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root ItemLimit4"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="8"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Some Hardware
        </p>
      </div>
      <span
        class="MuiTypography-root inStock MuiTypography-caption"
      >
        4 OF 6 IN STOCK
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InventoryItem limit 5 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root Item MuiGrid-item"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-outlined"
    >
      <img
        alt="Some Hardware"
        class="MuiCardMedia-root ItemPic MuiCardMedia-media MuiCardMedia-img"
        src="https://i.pinimg.com/originals/58/1f/23/581f232c3462392f63d57be5649649de.jpg"
      />
    </div>
    <div
      class="ItemTextBox"
    >
      <div
        class="ItemName"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root ItemLimit5"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="8"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Some Hardware
        </p>
      </div>
      <span
        class="MuiTypography-root inStock MuiTypography-caption"
      >
        4 OF 6 IN STOCK
      </span>
    </div>
  </div>
</DocumentFragment>
`;
